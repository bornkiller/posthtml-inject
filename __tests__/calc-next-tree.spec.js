/**
 * @description - main test suits
 * @author - bornkiller <hjj491229492@hotmail.com>
 */

// Internal
const calcNextTree = require('../lib/calc-next-tree');

describe('calc-next-tree helper', () => {
  it('should tolerate missing elements', () => {
    expect(calcNextTree({}, 'missing')).toEqual([]);
  });

  it('should tolerate incorrect elements syntax', () => {
    expect(calcNextTree({ incorrect: {} }, 'incorrect')).toEqual([]);
  });

  it('should normalize standard elements', () => {
    const elements = {
      manifest: [
        {
          tag: 'link',
          attrs: {
            href: 'manifest.json'
          }
        },
        {
          tag: 'link',
          attrs: {
            href: 'main.css'
          }
        }
      ]
    };

    expect(calcNextTree(elements, 'manifest', '\n', '  ')).toMatchSnapshot();
  });
});
