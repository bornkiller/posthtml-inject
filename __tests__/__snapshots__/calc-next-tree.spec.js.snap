// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calc-next-tree helper should normalize standard elements 1`] = `
Array [
  Object {
    "attrs": Object {
      "href": "manifest.json",
    },
    "tag": "link",
  },
  "
  ",
  Object {
    "attrs": Object {
      "href": "main.css",
    },
    "tag": "link",
  },
]
`;
